notebooks/llm-chatbot/ov_rerank_model.py

```diff
@@ -0,0 +1,162 @@
+
+    def __init__(self, query=None, passages=None):
+        self.query = query
+        self.passages = passages if passages is not None else []
+
+
+class OVRanker(BaseDocumentCompressor):
+    """
+    OpenVINO BGE rerank models.
+
+    """
+
+    ov_model: Any
+    """OpenVINO model object."""
+    tokenizer: Any
+    """Tokenizer for embedding model."""
+    model_dir: str
+    """Path to store models."""
+    model_kwargs: Dict[str, Any]
+    """Keyword arguments passed to the model."""
+    top_n: int = 4
+    """return Top n texts."""
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.tokenizer = self._get_tokenizer()
+        self.ov_model = OVModelForSequenceClassification.from_pretrained(
+            self.model_dir, **self.model_kwargs)
+
+    def _load_vocab(self, vocab_file):
+
+        vocab = collections.OrderedDict()
+        with open(vocab_file, "r", encoding="utf-8") as reader:
+            tokens = reader.readlines()
+        for index, token in enumerate(tokens):
+            token = token.rstrip("\n")
+            vocab[token] = index
+        return vocab
+
+    def _get_tokenizer(self, max_length=512):
+
+        config_path = Path(self.model_dir) / "config.json"
+        if not config_path.exists():
+          raise FileNotFoundError(f"config.json missing in {self.model_dir}")
+
+        tokenizer_path = Path(self.model_dir) / "tokenizer.json"
+        if not tokenizer_path.exists():
+          raise FileNotFoundError(
+              f"tokenizer.json missingin  {self.model_dir}")
+
+        tokenizer_config_path = Path(self.model_dir) / "tokenizer_config.json"
+        if not tokenizer_config_path.exists():
+          raise FileNotFoundError(
+              f"tokenizer_config.json missing in  {Path(self.model_dir)}")
+
+        tokens_map_path = Path(self.model_dir) / "special_tokens_map.json"
+        if not tokens_map_path.exists():
+          raise FileNotFoundError(
+              f"special_tokens_map.json missing in  {Path(self.model_dir)}")
+
+        config = json.load(open(str(config_path)))
+        tokenizer_config = json.load(open(str(tokenizer_config_path)))
+        tokens_map = json.load(open(str(tokens_map_path)))
+
+        tokenizer = Tokenizer.from_file(str(tokenizer_path))
+        tokenizer.enable_truncation(max_length=min(
+            tokenizer_config["model_max_length"], max_length))
+        tokenizer.enable_padding(
+            pad_id=config["pad_token_id"], pad_token=tokenizer_config["pad_token"])
+
+        for token in tokens_map.values():
+          if isinstance(token, str):
+              tokenizer.add_special_tokens([token])
+          elif isinstance(token, dict):
+              tokenizer.add_special_tokens([AddedToken(**token)])
+
+        vocab_file = Path(self.model_dir) / "vocab.txt"
+        if vocab_file.exists():
+          tokenizer.vocab = self._load_vocab(vocab_file)
+          tokenizer.ids_to_tokens = collections.OrderedDict(
```

Comment:
As I know regular dict in python preserves keys addition order starting from python3.6 (currently openvino supported minimal python version is 3.8), no need to use OrderedDict